core/src/client/async_client/helpers.rs

```diff
@@ -81,20 +81,15 @@ pub(crate) fn process_subscription_response(
 	let sub_id = response.params.subscription.into_owned();
 	let request_id = match manager.get_request_id_by_subscription_id(&sub_id) {
 		Some(request_id) => request_id,
-		None => return Err(None),
+		None => {
+			tracing::error!("Subscription ID: {:?} is not an active subscription", sub_id);
+			return Err(None);
+		}
 	};
 
 	match manager.as_subscription_mut(&request_id) {
-		Some(send_back_sink) => match send_back_sink.try_send(response.params.result.clone()) {
+		Some(send_back_sink) => match send_back_sink.try_send(response.params.result) {
 			// The server sent a subscription closed notification, then close down the subscription.
```

Comment:
I can't quite grasp the significance of removing this code (which was added fairly recently iirc to properly close a subscription)? Is this removed because we have removed any assumptions about how a subscription will be closed, and so it's up to the client to decide?

Reply:
(also I guess the comment can be removed too now?)

Reply:
So I had a gander at this code and I think it all makes sense to me; the backend sends a message, and we handle it in `process_subscription_response`, and here we are sending that message on to a channel held by somebody, anmd if the send is Ok then we just keep going, and if the send failed then the frontend has dropped the channel and we should end the subscription!

The only thing left here is to remove that comment then; we are no longer doing what it says :)